woocommerce/woocommerce-ios
# Product Creation AI: Tracking for main creation flow

Closes: #10795 Closes: #10809

## Description

- Adds tracking events for the main product creation AI flow. 
- Hides the feedback control after user provides feedback. 

## Testing instructions
- Log in to a WPCom or self-hosted store with Jetpack AI installed.
- Switch to the Products tab and select "+" to add a new product.
- Validate the following event is present in Xcode console
```
Tracked product_creation_ai_entry_point_displayed, properties: [AnyHashable("site_url"): <site_url>, AnyHashable("plan"): "jetpack_free", AnyHashable("blog_id"): <site_id>, AnyHashable("was_ecommerce_trial"): false, AnyHashable("is_wpcom_store"): false]
```
- Select add product with AI.
- Validate the following event is present in Xcode console
```
Tracked product_creation_ai_entry_point_tapped, properties: [AnyHashable("was_ecommerce_trial"): false, AnyHashable("blog_id"): <site_id>, AnyHashable("plan"): "jetpack_free", AnyHashable("site_url"): <site_url>, AnyHashable("is_wpcom_store"): false]
```
- Enter a product name -> Tap "Continue"
- Validate the following event is present in Xcode console
```
Tracked product_creation_ai_product_name_continue_button_tapped, properties: [AnyHashable("is_wpcom_store"): false, AnyHashable("plan"): "jetpack_free", AnyHashable("blog_id"): <site_id>, AnyHashable("was_ecommerce_trial"): false, AnyHashable("site_url"): <site_url>]
```
- Tap "Set tone and voice"
- Select a tone
- Validate the following event is present in Xcode console
```
Tracked product_creation_ai_tone_selected, properties: [AnyHashable("site_url"): <site_url>, AnyHashable("value"): "Formal", AnyHashable("is_wpcom_store"): false, AnyHashable("blog_id"): <site_id>, AnyHashable("was_ecommerce_trial"): false, AnyHashable("plan"): "jetpack_free"]
```
- Enter product features -> Tap "Create Product Details" button
- Validate the following event is present in Xcode console
```
Tracked product_creation_ai_generate_details_tapped, properties: [AnyHashable("is_wpcom_store"): false, AnyHashable("plan"): "jetpack_free", AnyHashable("blog_id"): <site_id>, AnyHashable("was_ecommerce_trial"): false, AnyHashable("site_url"): <site_url>, AnyHashable("is_first_attempt"): true]
```
- The `is_first_attempt` property should be false when trying the generation next time. (Navigate back from the preview screen and try again)
- Once product details are generated, you will see the product details in the Preview screen.
- Validate the following event is present in Xcode console
```
Tracked product_creation_ai_generate_product_details_success, properties: [AnyHashable("is_wpcom_store"): false, AnyHashable("site_url"): <site_url>, AnyHashable("was_ecommerce_trial"): false, AnyHashable("blog_id"): <site_id>, AnyHashable("plan"): "jetpack_free"]
```
- Go back to the features screen. 
- Turn off wifi ->  -> Tap "Create Product Details" button
- When the generation fails, you will see the following event in Xcode console
```
Tracked product_creation_ai_generate_product_details_failed, properties: [AnyHashable("site_url"):  <site_url>, AnyHashable("is_wpcom_store"): false, AnyHashable("error_domain"): "WooCommerce.IdentifyLanguageError", AnyHashable("was_ecommerce_trial"): false, AnyHashable("error_description"): "<error_details>", AnyHashable("blog_id"): <site_id>, AnyHashable("plan"): "jetpack_free", AnyHashable("error_code"): "0"]
```
- Enable wifi and generate details again
- Give feedback by tapping thumbs up or down and validate that the following event is tracked
```
Tracked product_ai_feedback, properties: [AnyHashable("is_useful"): true, AnyHashable("source"): "product_creation", AnyHashable("plan"): "ecommerce-trial-bundle-monthly", AnyHashable("is_wpcom_store"): true, AnyHashable("blog_id"): <site_id>, AnyHashable("was_ecommerce_trial"): true, AnyHashable("site_url"): <site_url>]
```
- Tap "Save as draft" button
- Validate the following event in Xcode console
```
Tracked product_creation_ai_save_as_draft_button_tapped, properties: [AnyHashable("site_url"): <site_url>, AnyHashable("blog_id"): <site_id>, AnyHashable("was_ecommerce_trial"): false, AnyHashable("is_wpcom_store"): false, AnyHashable("plan"): "jetpack_free"]
```
- Once the product is saved successfully you should see the following event in console
```
Tracked product_creation_ai_save_as_draft_success, properties: [AnyHashable("site_url"): <site_url>, AnyHashable("was_ecommerce_trial"): false, AnyHashable("plan"): "jetpack_free", AnyHashable("blog_id"): <site_id>, AnyHashable("is_wpcom_store"): false]
```
- When saving the product fails, you should see the following event in Xcode console
```
Tracked product_creation_ai_save_as_draft_failed, properties: [AnyHashable("is_wpcom_store"): false, AnyHashable("plan"): "jetpack_free", AnyHashable("blog_id"): 224075732, AnyHashable("error_code"): "0", AnyHashable("site_url"): <site_url>, AnyHashable("error_domain"): "Yosemite.ProductUpdateError", AnyHashable("was_ecommerce_trial"): false, AnyHashable("error_description"): "<error_details>"]
```


## Screenshots

https://github.com/woocommerce/woocommerce-ios/assets/524475/86fbe8c1-3a1b-41e1-9253-b44c527d0fc9



---
